web/api/v1/api.go

```diff
@@ -1567,6 +1610,47 @@ func parseExcludeAlerts(r *http.Request) (bool, error) {
 	return excludeAlerts, nil
 }
 
+func parseListRulesPaginationRequest(r *http.Request) (*listRulesPaginationRequest, *parsePaginationError) {
+	var (
+		parsedMaxGroups int64 = -1
+		err             error
+	)
+	maxGroups := r.URL.Query().Get("max_groups")
```

Comment:
What about `group_limit` naming instead? This might be consistent with other limit parameters in our API.

Reply:
I guess one counter argument is that `limit` never supported `tokens`/pagination, but maybe it does not matter and limit would be better?

Reply:
Consistency for the win :) . Actually can we say the APIs that have `limit` can support pagination in the future?